### PR TITLE
support chaining of ERB for Sprockets 4

### PR DESCRIPTION
**Pull request by ashkulz**
Also see PR to sprockets: rails/sprockets#807

### CODE REVIEW

**Comment by ashkulz** [comment]
@mathieujobin would appreciate your taking a look :slightly_smiling_face:

File: Gemfile [hunk 1]
```diff
@@ -2,6 +2,7 @@ source 'https://rubygems.org'
 
 gemspec
 
+gem "rack", "~> 2" # Rack 3 is not compatible with Jasmine and fails -- jasmine-3.99.0/lib/jasmine/server.rb:12:in `start': uninitialized constant Rack::Server (NameError)
```
Comment (mathieujobin) [inline Gemfile:5]: Oh, thank you

**Review by mathieujobin — APPROVED** [review]

File: lib/haml_coffee_assets/rails/engine.rb [hunk 2]
```diff
@@ -38,6 +38,9 @@ class Engine < ::Rails::Engine
             env.register_mime_type 'text/hamlc', extensions: ['.hamlc', '.jst.hamlc']
             env.register_transformer 'text/hamlc', 'application/javascript', ::HamlCoffeeAssets::Transformer
             env.register_transformer 'text/hamlc', 'application/javascript+function', ::HamlCoffeeAssets::Transformer
+            # support for chaining via ERB, documented via https://github.com/rails/sprockets/pull/807
+            env.register_mime_type 'text/hamlc+ruby', extensions: ['.hamlc.erb', '.jst.hamlc.erb']
+            env.register_transformer 'text/hamlc+ruby', 'text/hamlc', ::Sprockets::ERBProcessor
```
Comment (mathieujobin) [inline lib/haml_coffee_assets/rails/engine.rb:43]: This isn't exactly the same as #190 but equivalent right?

Comment (voducvu) [reply lib/haml_coffee_assets/rails/engine.rb:43]: They are different, @mathieujobin. #190 fixes hamlc without .erb

File: Gemfile (hunk 1 above)
Comment (voducvu) [reply Gemfile:5]: @ashkulz @mathieujobin  My team is using Rack 3. If your lock it at Rack 2 it would break our system

**Comment by ashkulz** [comment]
This is just used in tests, which use jasmine (now obsolete/ archived). There should be no downstream impact.

**Pull request merged**